client/src/pages/loginpage/LoginPage.jsx

```diff
@@ -15,22 +18,31 @@ export default function SignupPage() {
   } = useForm();
   const expressURL = import.meta.env.VITE_API_URL;
 
+  const redirect = () => {
+    if(resStatus === 200) navigate("/")
+  };
+
   const onSubmit = async (data) => {
     try {
       await axios
-        .post(`${expressURL}/api/users/login`, data, {
-          headers: { "Content-Type": "application/json" },
+        .post(`${expressURL}/api/auth/login`, data, 
+          {
+          withCredentials: true,
+        },
+          {
+          headers: { 'Content-Type' : 'application/x-www-form-urlencoded; charset=UTF-8' },
         })
-        .then((response) => setResStatus(response.status));
-      if (resStatus === 200) {
-        navigate("/");
-      }
+        .then((response) => {
+          setResStatus(parseInt(response.status, 10))
+          setCurrentUser(response.data.user);
+        }).finally(redirect())
```

Comment:
navigate here not redirect

Reply:
> navigate here not redirect

why should we use navigate instead of redirect?

Reply:
is checking for response status not necessary before redirecting or is there another issue ?